nistats/utils.py

```diff
@@ -40,6 +41,67 @@ def _check_and_load_tables(tables_, var_name):
                             (var_name, type(table), table_idx))
     return tables
 
+ 
+def _verify_events_file_uses_tab_separators(events_files):
+    """
+    Raises a ValueError if provided list of text based data files
+    (.csv, .tsv, etc) do not enforce the BIDS convention of using Tabs
+    as separators.
+    
+    Only scans their first row.
+    Does nothing if:
+        If the separator used is BIDS compliant.
+        Paths are invalid.
+        File(s) are not text files.
+
+    Does not flag comma-separated-values-files for compatibility reasons;
+    this may change in future as commas are not BIDS compliant.
+    
+    parameters
+    ----------
+    events_files: str, List/Tuple[str]
+        A single file's path or a collection of filepaths.
+        Files are expected to be text files.
+        Non-text files will raise ValueError.
+    
+    Returns
```

Comment:
Does it it need to return anything ?

Reply:
Not for functionality and I personally don't find this approach very elegant. I wanted to make sure I had a way to test the do-nothing cases and this was the most straightforward way. If I really wanted to, I suppose I should implement the logging module maybe.
If I do remove this, I will also remove the unit tests that depend on this. There's a consideration that those unit tests may not be necessary at all.
What do you think?

Reply:
For the tests, cant'you simply use assert_raises ?

Reply:
I think I get your point. However, I'm worried about API homogeneity: I would not like to introduce custom function signatures, simply to ensure that a unit test specifically tests one function. 
@GaelVaroquaux  any opinion on this ?

Reply:
I agree, this is rather inelegant; if there is a better way to do this I would like to know.
We should start introducing per function testing now, it will improve the code and coding practices by future contributors.

Reply:
I will remove the section returning the exceptions and modify the tests accordingly.
We should start insisting on er function unit tests from new contributors & contributions.